internal/pkg/es/error.go

```diff
@@ -66,12 +74,77 @@ var (
 	ErrIndexNotFound          = errors.New("index not found")
 	ErrTimeout                = errors.New("timeout")
 	ErrNotFound               = errors.New("not found")
+
+	knownErrorTypes = [3]string{
```

Comment:
nit: could omit the size (easier to add more elements later without updating the size)
```
knownErrorTypes = [...]string{
```